src/arraystring.rs

```diff
@@ -46,6 +47,16 @@ impl<const N: usize> ArrayString<N> {
         }
     }
 
+    #[inline]
+    fn remaining_mut(&mut self) -> &mut [u8] {
+        &mut self.array[self.size as usize..]
```

Comment:
We should not give access to invalid values in safe functions. If it's out of the valid string (0..len) it should only be accessible through unsafe functions.

Specially because we might use MaybeUninit now that is exists (at the time mem::uninitialized was too brittle)